vcd/resource_vcd_catalog_test.go

```diff
@@ -133,6 +133,163 @@ func TestAccVcdCatalog(t *testing.T) {
+		ProviderFactories: testAccProviders,
+		CheckDestroy:      testAccCheckCatalogDestroy,
+		Steps: []resource.TestStep{
+			// Test creation
+			{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+					cachedCatalogId.cacheTestResourceFieldValue(resourceCatalog, "id"),
+					cachedMediaId.cacheTestResourceFieldValue(resourceCatalog, "id"),
+					cachedvAppTemplateId.cacheTestResourceFieldValue(resourceCatalog, "id"),
+					cachedVMId1.cacheTestResourceFieldValue(resourceCatalog, "id"),
+					cachedVMId2.cacheTestResourceFieldValue(resourceCatalog, "id"),
+					testAccCheckVcdCatalogExists(resourceCatalog),
+					testAccCheckCatalogEntityState("vcd_catalog_media", orgName, catalogMediaName, true),
+					testAccCheckCatalogEntityState("vcd_catalog_vapp_template", orgName, vappTemplateName, true),
+					resource.TestCheckResourceAttr(resourceCatalog, "name", catalogName),
+					resource.TestCheckResourceAttr(resourceCatalog, "description", t.Name()),
+					resource.TestMatchResourceAttr(resourceCatalog, "catalog_version", regexp.MustCompile(`^\d+`)),
+					resource.TestMatchResourceAttr(resourceCatalog, "owner_name", regexp.MustCompile(`^\S+$`)),
+				),
+			},
+			{
+				// Intermediate step needed to rename the catalog before checking
+				// the vApp template and media depending on it.
+				Config: renameText,
+				Check: resource.ComposeTestCheckFunc(
+					cachedCatalogId.testCheckCachedResourceFieldValue(resourceCatalog, "id"),
+					cachedMediaId.testCheckCachedResourceFieldValue(resourceCatalog, "id"),
+					cachedvAppTemplateId.testCheckCachedResourceFieldValue(resourceCatalog, "id"),
+					cachedVMId1.testCheckCachedResourceFieldValue(resourceCatalog, "id"),
+					cachedVMId1.testCheckCachedResourceFieldValue(resourceCatalog, "id"),
+				),
+			},
+			{
+				Config: renameText,
+				Check: resource.ComposeTestCheckFunc(
+					cachedCatalogId.testCheckCachedResourceFieldValue(resourceCatalog, "id"),
+					cachedMediaId.testCheckCachedResourceFieldValue(resourceCatalog, "id"),
+					cachedvAppTemplateId.testCheckCachedResourceFieldValue(resourceCatalog, "id"),
+					cachedVMId1.testCheckCachedResourceFieldValue(resourceCatalog, "id"),
+					cachedVMId1.testCheckCachedResourceFieldValue(resourceCatalog, "id"),
+					resource.TestCheckResourceAttr(resourceCatalog, "name", catalogUpdatedName),
+					resource.TestCheckResourceAttr(resourceCatalog, "number_of_vapp_templates", "1"),
+					resource.TestCheckResourceAttr(resourceCatalog, "number_of_media", "1"),
+					resource.TestCheckResourceAttrPair(resourcevAppTemplate, "catalog_id", resourceCatalog, "id"),
+					resource.TestCheckResourceAttrPair(resourceMedia, "catalog_id", resourceCatalog, "id"),
+				),
+			},
+		},
+	})
+}
+
+const testAccCheckVcdCatalogRename = `
+resource "vcd_catalog" "test-catalog" {
+  org = "{{.Org}}" 
+  
+  name        = "{{.CatalogName}}"
+  description = "{{.Description}}"
+
+  delete_force      = "true"
+  delete_recursive  = "true"
```

Comment:
```suggestion
  delete_force     = "true"
  delete_recursive = "true"
```

Embedded HCL scripts should be well formatted

Reply:
Fixed.